### PR TITLE
Using a custom transmission for biotacs to solve the j0 issue in Gazebo

### PR DESCRIPTION
**Pull request by ugocupcic**

### CODE REVIEW

**Comment by ugocupcic** [comment]
still need to use it in the sr_description.

File: sr_mechanism_model/src/biotac_transmission.cpp [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+    {
+      ROS_ERROR("BiotacTransmission could not find actuator named \"%s\"", actuator_name);
+      return false;
+    }
+    actuator_names_.push_back(actuator_name);
+
+    mechanical_reduction_ = atof(elt->FirstChildElement("mechanicalReduction")->GetText());
+
+    return true;
+  }
```
Comment (shadowmanos) [inline sr_mechanism_model/src/biotac_transmission.cpp:98]: initXml(TiXmlElement *elt) can call initXml(TiXmlElement *elt, pr2_mechanism_model::Robot *robot) with NULL as the second argument. Then only the if statement and a->command_.enable_ = true; need a little modifying and a lot of code duplication would be avoided

File: sr_mechanism_model/src/biotac_transmission.cpp [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        js[1]->position_ = static_cast<sr_actuator::SrActuator*>(as[0])->state_.calibrated_sensor_values_[1];
+
+        js[0]->velocity_ = static_cast<sr_actuator::SrActuator*>(as[0])->state_.velocity_ / 2.0;
+        js[1]->velocity_ = static_cast<sr_actuator::SrActuator*>(as[0])->state_.velocity_ / 2.0;
+
+        js[0]->measured_effort_ = static_cast<sr_actuator::SrActuator*>(as[0])->state_.last_measured_effort_;
+        js[1]->measured_effort_ = static_cast<sr_actuator::SrActuator*>(as[0])->state_.last_measured_effort_;
+      }
+    }
+    else
```
Comment (shadowmanos) [inline sr_mechanism_model/src/biotac_transmission.cpp:152]: if (size == 2)
{
}
else if (size == 0)
{
}

might be more concise here

File: sr_mechanism_model/src/biotac_transmission.cpp [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+      static_cast<sr_actuator::SrActuator*>(as[0])->state_.sample_timestamp_ = ros::Time::now() - simulated_actuator_start_time_;
+    }
+    // Set the historical (double) timestamp accordingly.
+    static_cast<sr_actuator::SrActuator*>(as[0])->state_.timestamp_ = static_cast<sr_actuator::SrActuator*>(as[0])->state_.sample_timestamp_.toSec();
+
+    // simulate calibration sensors by filling out actuator states
+    this->joint_calibration_simulator_.simulateJointCalibration(js[0],static_cast<sr_actuator::SrActuator*>(as[0]));
+
+    ROS_DEBUG(" end propagate pos backward for j0");
+  }
```
Comment (shadowmanos) [inline sr_mechanism_model/src/biotac_transmission.cpp:216]: The function above would be more concise if something like
sr_actuator::SrActuator\* actuator = static_cast&lt;sr_actuator::SrActuator*&gt;(as[0]);
was defined at the beginning and used afterwards.

**Pull request closed without merging**